src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -256,10 +262,25 @@ private static String getSkippedMessage(Element testCase) {
         return message;
     }
 
-    public String getDisplayName() {
+    public String getTransformedTestName() {
         return TestNameTransformer.getTransformedName(testName);
     }
 
+    public String getDisplayName() {
+        // Only prepend the enclosing flow node names if there are any and the run this is in has multiple blocks directly containing
+        // test results.
+        if (!getEnclosingFlowNodeNames().isEmpty()) {
```

Comment:
So previous node here means `parents`? 

We're kinda stuck using `FlowNode`s for `getEnclosingBlocks()` (rather than string IDs) - well, more accurately, we're going to have to load the `FlowNode` for the immediate enclosing ID to be able to get *its* enclosing block and so on, so if we're going that far, hey, why not just stick that all in a `transient` field and avoid having to do that loading again later.